pkg/kubelet/cm/deviceplugin/endpoint.go

```diff
@@ -77,24 +78,22 @@ func (e *endpointImpl) callback(resourceName string, added, updated, deleted []p
 	e.cb(resourceName, added, updated, deleted)
 }
 
-func (e *endpointImpl) getDevices() []pluginapi.Device {
+// getDevices makes a deep copy of the ID->Device mapping
+func (e *endpointImpl) getDevices() map[string]pluginapi.Device {
```

Comment:
I've already addressed cleanup of the Device interface in PR #52616

Reply:
Right, I should have spent some time on that PR. I went sleepless last night thinking of ways to get the concurrency management do right. The data structure has to be correct or else all the efforts on getting mutex/rwmutex done right will go in vain.

Reply:
Agreed current concurrency management is a bit nuts...